src/asl.ts

```diff
@@ -2775,12 +2805,8 @@ export class ASL {
           typeof constant.constant === "object" &&
           constant.constant !== null
         ) {
-          throw new SynthError(
-            ErrorCodes.StepFunction_invalid_filter_syntax,
-            `Filter expressions do not support object or array literals, found: ${JSON.stringify(
-              constant.constant
-            )}`
-          );
+          // will be caught, try to compile with ASLGraph instead
+          throw new FilterJsonPathUnsupported("Use ASLGraph");
```

Comment:
Is it possible to have a functional API that doesn't depend on throwing this to signal that ASLGraph should be used? return `undefined`?

Reply:
Not in a clean way, I'd also prefer that, but this method is always embedded in string templates `something.${filterToJsonPath()}`. Would need to re-rewrite them all to call the method and check for null and then chain it down. This was the lowest touch and the error is always caught.

Reply:
I'll play with it a bit.

Reply:
just made it monadic with `string | undefined`. accident `undefined` in string templates scares me, but I think I got them all.